### PR TITLE
feat: setup supabase middleware

### PR DESCRIPTION
**Pull request by jeonghoyang12**

### CODE REVIEW

File: src/db/supabase/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+export async function updateSession(request: NextRequest) {
+  let response = NextResponse.next({
+    request: {
+      headers: request.headers,
+    },
+  })
+
+  const supabase = createServerClient(
+    process.env.NEXT_PUBLIC_SUPABASE_URL!,
```
Comment (devjiwonchoi) [inline src/db/supabase/middleware.ts]: what if we throw early if not exist?

File: src/db/supabase/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  let response = NextResponse.next({
+    request: {
+      headers: request.headers,
+    },
+  })
+
+  const supabase = createServerClient(
+    process.env.NEXT_PUBLIC_SUPABASE_URL!,
+    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
```
Comment (devjiwonchoi) [inline src/db/supabase/middleware.ts]:
```suggestion
    process.env.NEXT_PUBLIC_SUPABASE_URL,
    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY,
```

**Review by devjiwonchoi — CHANGES_REQUESTED** [review]

**Review by devjiwonchoi — APPROVED** [review]
Lgtm, just so you know, middleware runs on EVERY route, should consider if its necessary in our case!

**Comment by jeonghoyang12** [comment]
> Lgtm, just so you know, middleware runs on EVERY route, should consider if its necessary in our case!

**Comment by jeonghoyang12** [comment]
> Lgtm, just so you know, middleware runs on EVERY route, should consider if its necessary in our case!

Good point, will just close this PR for now, and will consider adding middleware when we actually need this.

**Pull request closed without merging**